### PR TITLE
[receiver/mongodbreceiver] Partial error fix

### PR DESCRIPTION
**Pull request by JonathanWamsley**
**Description:** 
- fixes issues where partial errors are not recorded due to passing the ScrapeErrors struct without a pointer
- rename errors/scraperErrors to errs to avoid possible shadowing of the errors `package`

**Link to tracking Issue:** 
#13108

### CODE REVIEW

File: receiver/mongodbreceiver/README.md [hunk 1]
```diff
@@ -61,6 +60,13 @@ The full list of settings exposed for this receiver are documented [here](./conf
 
 ## Metrics
 
+The following metric are available with versions:
+- `mongodb.extent.count` < 4.4 with mmapv1 storage engine
+- `mongodb.session.count` >= 3.0 with wiredTiger storage engine
+- `mongodb.cache.operations` >= 3.0 with wiredTiger storage engine
+- `mongodb.connection.count` with attribute `active` >= 4.0
+- `mongodb.index.access.count` >= 4.0
+
```
Comment (djaglowski) [inline receiver/mongodbreceiver/README.md:69]: This is a nice improvement, but it's unrelated to fixing the partial errors. Please submit as a separate PR.

File: receiver/mongodbreceiver/testdata/integration/expected.3_0.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
                         "dataPoints": [
                            {
                               "asInt": "0",
-                              "startTimeUnixNano": "1659372711430846000",
-                              "timeUnixNano": "1659372771440281000"
+                              "startTimeUnixNano": "1659728796441872000",
```
Comment (djaglowski) [inline receiver/mongodbreceiver/testdata/integration/expected.3_0.json:16]: Is this file changed because fixing the partial error problem changed the result?

File: receiver/mongodbreceiver/metrics.go [hunk 3]
```diff
@@ -44,142 +44,152 @@ var documentMap = map[string]metadata.AttributeOperation{
...
-		return
-	}
-	extentsVal, err := parseInt(extents)
-	if err != nil {
-		errors.AddPartial(1, err)
-		return
+func (s *mongodbScraper) recordExtentCount(now pcommon.Timestamp, doc bson.M, dbName string, errs *scrapererror.ScrapeErrors) {
+	// Mongo version 4.4+ no longer returns numExtents since it is part of the obsolete MMAPv1
+	// https://www.mongodb.com/docs/manual/release-notes/4.4-compatibility/#mmapv1-cleanup
+	mongo44, _ := version.NewVersion("4.4")
```
Comment (djaglowski) [inline receiver/mongodbreceiver/metrics.go:140]: Please submit these version-related changes in a separate PR.

File: receiver/mongodbreceiver/scraper_test.go [hunk 4]
```diff
@@ -118,37 +115,6 @@ func TestScrapeNoClient(t *testing.T) {
 	require.Error(t, err)
 }
 
-func TestGlobalLockTimeOldFormat(t *testing.T) {
```
Comment (djaglowski) [inline receiver/mongodbreceiver/scraper_test.go:121]: Why is this removed?

File: receiver/mongodbreceiver/README.md (hunk 1 above)
Comment (JonathanWamsley) [reply receiver/mongodbreceiver/README.md]: will add in follow up pr

File: receiver/mongodbreceiver/testdata/integration/expected.3_0.json (hunk 2 above)
Comment (JonathanWamsley) [reply receiver/mongodbreceiver/testdata/integration/expected.3_0.json]: this file does not need to be updated and for this pr

File: receiver/mongodbreceiver/metrics.go (hunk 3 above)
Comment (JonathanWamsley) [reply receiver/mongodbreceiver/metrics.go]: will do

**Review by djaglowski — APPROVED** [review]
